src/inference/src/dev/core_impl.cpp

```diff
@@ -1694,6 +1733,11 @@ std::string ov::CoreConfig::get_cache_dir() const {
     return _cacheConfig._cacheDir;
 }
 
+ov::CacheMode ov::CoreConfig::get_cache_mode() const {
```

Comment:
Caching related changes should be tested e.g.:
src/inference/tests/functional/caching_test.cpp

here as common tests on all devices
src/tests/functional/base_func_tests/include/behavior/compiled_model/compiled_model_base.hpp